docs/source/advanced/upload.mdx

```diff
@@ -2,24 +2,26 @@
 title: Uploading files
 ---
 
-Apollo Android supports file uploading over [graphql-multipart-request-spec](https://github.com/jaydenseric/graphql-multipart-request-spec).
+Apollo Android supports file uploading following
+the [GraphQL multipart request specification](https://github.com/jaydenseric/graphql-multipart-request-spec).
 
-You need to define this mapping in your build.gradle file.
+First, add the following custom scalar mapping to the Apollo Gradle plugin configuration:
 
-```groovy
+```kotlin:title=build.gradle.kts
 apollo {
-  customScalarsMapping = [
-    "Upload" : "com.apollographql.apollo3.api.Upload"
-  ]
+  customScalarsMappingset(mapOf(
+    "Upload" to "com.apollographql.apollo3.api.Upload"
+  ))
```

Comment:
I think most of Android projects (unfortunately) still use Groovy build scripts... Would be nice to keep Groovy around. If we can't find a syntax that works for both, maybe add `MultiCodeBlock`?